### PR TITLE
Java: add GETDEL command

### PR DESCRIPTION
**Pull request by cyip10**
https://redis.io/docs/latest/commands/getdel/

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 1]
```diff
@@ -34,6 +34,26 @@ public interface StringBaseCommands {
      */
     CompletableFuture<String> get(String key);
 
+    /**
+     * Get the value associated with the given <code>key</code>. <code>GETDEL</code> is similar to
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:38]:
```suggestion
     * Gets a string value associated with the given <code>key</code> and deletes the key. 
```

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 2]
```diff
@@ -34,6 +34,26 @@ public interface StringBaseCommands {
...
+     * @see <a href="https://redis.io/docs/latest/commands/getdel/">redis.io</a> for details.
+     * @param key The <code>key</code> to retrieve from the database.
+     * @return If <code>key</code> exists, returns the <code>value</code> of <code>key</code> as a
+     *     <code>String</code>. Otherwise, return <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * String value = client.getdel("key").get();
+     * assert value.equals("value");
+     *
+     * String value = client.get("key").get();
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:51]: nit
```suggestion
     * String value = client.getdel("key").get();
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 3]
```diff
@@ -305,6 +306,22 @@ public T get(@NonNull String key) {
         return getThis();
     }
 
+    /**
+     * Get the value associated with the given <code>key</code>. <code>GETDEL</code> is similar to
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:310]: Copy updated doc here.

File: java/gradlew.bat
Comment (Yury-Fridlyand) [inline java/gradlew.bat:1]: OOps - revert

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 4]
```diff
@@ -226,6 +226,25 @@ public void get_requires_a_key(BaseClient client) {
...
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void getdel(BaseClient client) {
+        String key1 = "{key}" + UUID.randomUUID();
+        String value1 = String.valueOf(UUID.randomUUID());
+        String key2 = "{key}" + UUID.randomUUID();
+
+        client.set(key1, value1).get();
+        String data = client.getdel(key1).get();
```
Comment (Yury-Fridlyand) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:238]: Repeat `getdel` on deleted `key1` to show that it returns `null`.

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java (hunk 2 above)
Comment (cyip10) [reply java/client/src/main/java/glide/api/commands/StringBaseCommands.java:51]: @Yury-Fridlyand Here i am trying to show that if they call get again, they will receive a null value

Comment (cyip10) [reply java/client/src/main/java/glide/api/commands/StringBaseCommands.java:51]: oh wait but i can use getdel again here anyways

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 5]
```diff
@@ -305,6 +306,22 @@ public T get(@NonNull String key) {
...
 
+    /**
+     * Gets a string value associated with the given <code>key</code> and deletes the key. {@see
+     * #get}, except for the fact that it also deletes the key on success (if and only if the value of
+     * the <code>key</code> is a string).
+     *
+     * @see <a href="https://redis.io/docs/latest/commands/getdel/">redis.io</a> for details.
+     * @param key The <code>key</code> to retrieve from the database.
+     * @return Command Response - If <code>key</code> exists, returns the <code>value</code> of <code>
+     *     key</code> as a <code>String</code>. Otherwise, return <code>null</code>.
```
Comment (GumpacG) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:317]:
```suggestion
     * @return Command Response - If <code>key</code> exists, returns the <code>value</code> of <code>
     *     key</code>. Otherwise, return <code>null</code>.
```

File: java/gradlew.bat [hunk 6]
```diff
@@ -1,92 +1,92 @@
-@rem
```
Comment (GumpacG) [inline java/gradlew.bat:1]: Unrelated changes?

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 7]
```diff
@@ -34,6 +34,26 @@ public interface StringBaseCommands {
      */
     CompletableFuture<String> get(String key);
 
+    /**
+     * Gets a string value associated with the given <code>key</code> and deletes the key. {@see
+     * #get}, except for the fact that it also deletes the key on success (if and only if the value of
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:39]: Not sure that we need the second phrase. @aaron-congo, what do you think?

File: java/gradlew.bat
Comment (Yury-Fridlyand) [inline java/gradlew.bat:1]: Revert

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java (hunk 7 above)
Comment (aaron-congo) [reply java/client/src/main/java/glide/api/commands/StringBaseCommands.java:39]: I think we can remove it

File: java/client/src/main/java/glide/api/commands/StringBaseCommands.java [hunk 8]
```diff
@@ -34,6 +34,25 @@ public interface StringBaseCommands {
      */
     CompletableFuture<String> get(String key);
 
+    /**
+     * Gets a string value associated with the given <code>key</code> and deletes the key. {@see
+     * #get}.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/StringBaseCommands.java:39]: I think we can remove the pointer to GET, same applies for the Javadocs in BaseTransaction
```suggestion
     * Gets a string value associated with the given <code>key</code> and deletes the key.
```

**Review by Yury-Fridlyand — APPROVED** [review]
LGTM assuming you revert `gradlew.bat`
